### ISSUE TITLE
Configure approving PRs for Dependabot runs 

### ISSUE THREAD

**Issue by rajbos** [devops-actions/actionlint#15]

**Comment by kevinmatthes** [comment]
[Renovate](https://github.com/renovatebot/renovate) is another GitHub bot which also suggests updates for dependencies as Dependabot does but in addition, it can also automatically merge its own PRs, if the respective boolean configuration variable is set to true.  There is an official approval bot available, too, which will automatically approve Renovate's PRs in case any branch protection rule should require this.

Furthermore, Renovate is not limited to the registries Dependabot supports but also allows for the definition of custom dependencies using regular expressions:  you define the regex(es) in a list of files to maintain as well as a repository to listen to for GitHub releases such that, if that repository receives a new GitHub release, Renovate will suggest the respective updates.  According to the documentation, this shall also work for commit SHAs.

The smallest time interval Dependabot allows for is daily, Renovate defaults to immediately -- whenever an update of a detected dependency is published, Renovate will instantly suggest the corresponding PR.  The configuration for the update intervals is more flexible than Dependabot's and can be also given in plain English.

Maybe Renovate might be an option to enhance the self-maintenance of this repository as the configuration of multiple update processes would be bundled into one file which can be also centralised?

**Comment by kevinmatthes** [comment]
@rajbos

This is the documentation page for updating Docker dependencies with Renovate:  https://github.com/renovatebot/renovate/blob/main/docs/usage/docker.md.  There is also a section on digest pinning, so there definitely is a way to have Renovate updating the called Docker image.  Please note that Renovate will keep the version tag for readability; see https://github.com/renovatebot/renovate/blob/main/docs/usage/docker.md?plain=1#L91.

One option would be to have an unused Dockerfile which is maintained by Renovate and only contains the actionlint requirement:

```dockerfile
FROM rhysd/actionlint:1.6.25@sha256:2595154b67df16b8bb9cf1ca01eeefe01bc18b68daa5ee745ba8edf2405be740
```

In the Action, you could then extract the Docker image using `cat Dockerfile | cut -d\  -f2`.

What do you think about this procedure?
